src/main/scala/com/microsoft/hyperspace/index/IndexDataManager.scala

```diff
@@ -47,17 +47,19 @@ class IndexDataManagerImpl(indexPath: Path) extends IndexDataManager {
   // TODO: Investigate whether FileContext should be used instead of FileSystem for atomic renames.
   private lazy val fs: FileSystem = indexPath.getFileSystem(new Configuration)
 
+  private lazy val absoluteIndexPath: Path = fs.makeQualified(indexPath)
```

Comment:
How about we move this to the caller side (`IndexLogManagerFactoryImpl` and `IndexDataManagerFactoryImpl`) and just do `assert(indexPath.isAbsolute)` in this class?

Reply:
thanks @imback82 , done